### PR TITLE
WS-1518: remove guava from rosette-api

### PR DESCRIPTION
**Pull request by Katsuya-Tomioka**

### CODE REVIEW

File: api/src/main/java/com/basistech/rosette/api/HttpRosetteAPI.java [hunk 1]
```diff
@@ -159,7 +159,19 @@ private static String getVersion() {
...
-        return ByteStreams.toByteArray(is);
+        ByteArrayOutputStream out = new ByteArrayOutputStream();
+
+        byte[] buf = new byte[4096];
+        long total = 0;
+
+        while (true) {
+            int r = is.read(buf);
+            if (r == -1) {
+                return out.toByteArray();
```
Comment (guruxu) [inline api/src/main/java/com/basistech/rosette/api/HttpRosetteAPI.java:170]: you may want to flush it first

File: api/src/main/java/com/basistech/rosette/api/HttpRosetteAPI.java [hunk 2]
```diff
@@ -159,7 +159,19 @@ private static String getVersion() {
...
+            }
+            out.write(buf, 0, r);
+            total += (long)r;
```
Comment (guruxu) [inline api/src/main/java/com/basistech/rosette/api/HttpRosetteAPI.java]: anyone needs a total?

**Comment by guruxu** [comment]
There an a few more guava consumption in tests, maybe we don't care but I don't know how guava wound up available there now that you removed the dependency.

There's also an unused(?) guava in `examples`?

We should try to get this into the 18.12 release, if possible

**Review by guruxu — APPROVED** [review]
LGTM

**Comment by Katsuya-Tomioka** [comment]
guava in tests are not good. I'll fix examples.

**Comment by Katsuya-Tomioka** [comment]
rosette-api-json does require guava for adm-json, only in deserialization. We might want to fix this -- though  this won't be an issue as long as "outside" doesn't use JSON -> Java conversion, I think. (This happens in "inside" once received a request.
Tests here are fine, using guava is fine as is.
examples needs guava as transitive.

**Comment by guruxu** [comment]
As long as we are not explicitly using guava directly here, having it in adm-json is fine.

**Pull request merged**